spot_rl_experiments/spot_rl/utils/skill_executor.py

```diff
@@ -31,6 +37,18 @@ def reset_skill_name_input(self, skill_name, succeded, msg):
         rospy.set_param("/skill_name_input", "None,None")
         rospy.set_param("/skill_name_suc_msg", f"{skill_name},{succeded},{msg}")
 
+    def read_emergency_stop(self):
+        while True:
+            estop = rospy.get_param("estop", False)
+            if estop:
+                print("Emergency Stop!!! Robot returns to dock")
+                spotskillmanager = SpotSkillManager(
+                    use_mobile_pick=True, use_semantic_place=True
+                )
+                spotskillmanager.dock()
+                self.end = True
+                raise SystemExit
```

Comment:
Hi Jimmy,

I think what I was thinking for e-stop is that we can just let robot "settle down" like with do in `spot_estop` functionality.

If we raise errors then we are killing out the node which is not necessarily what we want to do?

Want to know your thoughts.